### PR TITLE
celinanperalta/OAP 65

### PR DESCRIPTION
**Pull request by celinanperalta**
- Refactor: ngrams and suggestions tables now include more details about OAPEN items such as name and thumbnail link
- Refactor: oapen_types now uses typing.NamedTuple classes for cleaner attribute access
- Removed `suggestion` postgres type from schema. Instead of one record being (handle, suggestions[]), it is now (handle, suggestion_handle, ...). Essentially, each record is constrained by a unique item/suggestion pair.
- Removed get_collection_by_label and corresponding test
- Improvement: Randomly rotate User Agents when making requests to the OAPEN API, as well as randomly delay new requests between 1-10 seconds.

Downstreams affected:
- May need to fix embed and API querying given the new schema

### CODE REVIEW

File: oapen-engine/src/data/oapen_db.py [hunk 1]
```diff
@@ -93,11 +106,11 @@ def add_single_suggestion(self, suggestion: SuggestionRow) -> None:
     def add_many_suggestions(self, suggestions: List[SuggestionRow]) -> None:
         cursor = self.connection.cursor()
         args = self.mogrify_suggestions(suggestions)
+
         query = f"""
-                INSERT INTO oapen_suggestions.suggestions (handle, name, suggestion, score)
```
Comment (justinoboyle) [inline oapen-engine/src/data/oapen_db.py:97]: didnt know you could do this!

**Review by justinoboyle — APPROVED** [review]

**Comment by justinoboyle** [comment]
DB flattening is great. Looks good to me

**Comment by celinanperalta** [comment]
Hold on merging main messed up everything marking as draft until fixed :^)

**Comment by celinanperalta** [comment]
This is ready to be merged pending the resolution of the weird connection dropping on main

**Pull request merged**